### ISSUE TITLE
Full disk encryption

### ISSUE THREAD

**Issue by mpodshivalin** [bengtfredh/pinebook-pro-fedora-installer#28]
The install script is awesome, thank you for your work! For me, Fedora is the most usable OS on the Pinebook since I prefer GNOME, and Fedora ships with stock GNOME without customizations which make GNOME look like KDE :)

However, one thing that keeps me from using it on a daily basis is the absense of the full disk encryption - I carry a lot of SSH keys that I really can't afford to lose.

It seems that Full disk encryption should be possible, at least "similar" install script for Debian can do that https://github.com/daniel-thompson/pinebook-pro-debian-installer/

Is it possible to add that option?

**Comment by bengtfredh** [comment]
Thank you. It is a good idea to add encryption. I will look in to it when I have som free time.

**Comment by mpodshivalin** [comment]
Another example with FDE - [Manjaro ARM installer script](https://gitlab.manjaro.org/manjaro-arm/applications/manjaro-arm-installer/-/blob/master/manjaro-arm-installer)

**Comment by fredhas** [comment]
I am sorry no update yet. I have looked at it, and should not be to hard. Just been busy in my dayjob.  

Peter Robinson have patched u-boot for Fedora so it is possible to install official Fedora on the Pinebook Pro now. I have not tried it yet. But I guess you should be able to follow his tutorial to write firmware to flash (https://nullr0ute.com/2021/05/fedora-on-the-pinebook-pro/). You may be able to use the aarch64 installer image (https://download.fedoraproject.org/pub/fedora/linux/releases/34/Everything/aarch64/iso/Fedora-Everything-netinst-aarch64-34-1.2.iso) and be able to select to encrypt the drive during installation.  

Please let me know if you go down this route and if it works.

**Comment by bengtfredh** [comment]
I am sorry no update yet. I have looked at it, and should not be to hard. Just been busy in my dayjob.

Peter Robinson have patched u-boot for Fedora so it is possible to install official Fedora on the Pinebook Pro now. I have not tried it yet. But I guess you should be able to follow his tutorial to write firmware to flash (https://nullr0ute.com/2021/05/fedora-on-the-pinebook-pro/). You may be able to use the aarch64 installer image (https://download.fedoraproject.org/pub/fedora/linux/releases/34/Everything/aarch64/iso/Fedora-Everything-netinst-aarch64-34-1.2.iso) and be able to select to encrypt the drive during installation.

Please let me know if you go down this route and if it works.

**Comment by mpodshivalin** [comment]
So far I've managed to:
1) Write the firmware to flash (had to disable eMMC for that)
2) Download the installer Fedora-Everything-netinst-aarch64-34-1.2.iso, write the installer image to SD card (but using the standard Disk Image Writer, because it seems that arm-image-installer doesn't accept iso images)
3) Boot into the SD card using `devnum=1; run mmc_boot` at boot
4) Get the standard Fedora installation dialog, select "Install Fedora 34"

Then there is a message:
```
EFI stub: Booting Linux Kernel...
EFI stub: Using DTB from configuration table
EFI stub: Exiting boot services and installing virtual address map...
```

But then, unfortunately, there is only a black screen.

"Install Fedora 34 in basic graphics mode" also gives the same results.

Writing Fedora-Workstation-34-1.2.aarch64.raw.xz with arm-image-installer does produce a working image though

**Comment by bengtfredh** [comment]
Thank you for your update. This is pretty much a summary of what I experienced when I tried this weekend. I had high hopes after reading Peter Robinson blog. I was not able to boot from usb as he state in his blog to. Did you try to use the Workstation image? I had some problem booting, looks like Pinebook go in to maskrom mode every time I shut down. I started on a branch with encryption, it is not ready. It do not boot by now, I need to figure out why. I was thinking if I should rewrite the script to do it the Fedora way with grub efi boot and Fedora kernel.

**Comment by mpodshivalin** [comment]
Yes, for me Fedora-Workstation-34-1.2 worked (without NVMe and without WiFi but Peter mentioned that problem). One idea on how to shoot myself in the foot is to try to install calamares (graphic installer) when booted from an SD and then try to install Fedora to eMMC but I haven't tried that yet :)
